### PR TITLE
Updated crashlytics upload action to support multiple dSYM paths

### PR DESCRIPTION
**Pull request by KrauseFx**
This will allow us to use the dSYM files from the `Actions.lane_context[SharedValues::DSYM_PATHS` variable which is generated by `download_dsyms`

### CODE REVIEW

File: fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb [hunk 1]
```diff
@@ -14,11 +14,17 @@ def self.run(params)
         end
         UI.user_error!("Please provide an api_token using api_token:") unless params[:api_token]
 
-        command = []
-        command << File.expand_path(params[:binary_path])
-        command << "-a #{params[:api_token]}"
-        command << "-p #{params[:platform]}"
-        command << File.expand_path(params[:dsym_path])
+        dsym_paths = []
+        dsym_paths << params[:dsym_path]
```
Comment (lacostej) [inline fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb]: does that mean that the dsym_path param could become optional ? could it be skipped if nil ?

Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb]: Nice catch, the `dsym_path` parameter should be optional then.

File: fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb [hunk 2]
```diff
@@ -14,11 +14,17 @@ def self.run(params)
...
+
+        dsym_paths.each do |current_path|
+          command = []
+          command << File.expand_path(params[:binary_path])
+          command << "-a #{params[:api_token]}"
+          command << "-p #{params[:platform]}"
+          command << File.expand_path(current_path)
+        end
 
         return Actions.sh command.join(" ")
```
Comment (lacostej) [inline fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb]: don't you want to run the action multiple times ? Have you tried this ?

**Comment by KrauseFx** [comment]
@lacostej I updated the PR, also added a message if no files were found.

File: fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb [hunk 3]
```diff
@@ -14,13 +14,23 @@ def self.run(params)
...
+        dsym_paths.each do |current_path|
+          command = []
+          command << File.expand_path(params[:binary_path])
+          command << "-a #{params[:api_token]}"
+          command << "-p #{params[:platform]}"
+          command << File.expand_path(current_path)
+          results << Actions.sh(command.join(" "))
+        end
+
+        return results
```
Comment (lacostej) [inline fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb]: out of curiosity, what are the conventions for return values in actions ? Given that now we go from value to []

File: fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb [hunk 4]
```diff
@@ -14,13 +14,23 @@ def self.run(params)
...
-        command << File.expand_path(params[:binary_path])
-        command << "-a #{params[:api_token]}"
-        command << "-p #{params[:platform]}"
-        command << File.expand_path(params[:dsym_path])
+        dsym_paths = []
+        dsym_paths << params[:dsym_path]
+        dsym_paths += Actions.lane_context[SharedValues::DSYM_PATHS] if Actions.lane_context[SharedValues::DSYM_PATHS]
 
-        return Actions.sh command.join(" ")
+        UI.error("Couldn't find any dSYMs, please pass them using the dsym_path option") if dsym_paths.count == 0
```
Comment (lacostej) [inline fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb]: user_error! ?

File: fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb (hunk 3 above)
Comment (lacostej) [inline fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb]: other question is does upload-symbols support multiple dDSYM in one call ?

File: fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb (hunk 4 above)
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb]: It shouldn't fail if there is no dSYM file, we just want to tell the user

File: fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb (hunk 3 above)
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb]: This one is really only for running tests properly, I should fix that

**Comment by asfalcone** [comment]
LGTM! :arrow_double_up:

**Pull request merged**